File: frontend/src/types/Order.ts
export interface CreateOrderDTO {
  items: {
    id: string;
    quantity: number;
  }[];
  user: {
    email: string;
    phoneNumber: string;
    address: string;
  };
  neighborhoodId: string;
  paymentMethod: PaymentMethods;
  observation?: string | undefined;
}

export enum PaymentMethods {
  Pix = "PIX",
  CartaoDebito = "DEBIT_CARD",
  CartaoCredito = "CREDIT_CARD",
}

File: backend/src/types/order.py
from uuid import UUID
from pydantic import BaseModel, EmailStr

from src.domain.shared.phone_number import PhoneNumber
from src.domain.order import PaymentMethod


class CreateOrderRequestItem(BaseModel):
    product_id: UUID
    quantity: int


class CreateOrderRequestUserDTO(BaseModel):
    email: EmailStr
    phone: PhoneNumber
    address: str


class CreateOrderRequestDTO(BaseModel):
    restaurant_id: UUID
    items: list[CreateOrderRequestItem]
    user: CreateOrderRequestUserDTO
    payment_method: PaymentMethod
    observation: str = ""
